Docu

dkorolev opened the pull request:
Preparing for the 1.0 launch.

Major changes:

1. Renamed `JSONParse()` into `ParseJSON()`.
2. Flipped the arguments for `WriteStringToFile()` to be string first, then file.
3. `HTTP` and `cerealize` major usecases no longer require `using`-s, they are in a global namespace.

Minor changes, mostly in the documentation.

* Added documentation for `gnuplot` and `plotutils` (depends on https://github.com/KnowSheet/Bricks/pull/70)
* Moved RTTI's documentation to be the last one.
* Fixed a couple links, phrasings and comments in the documentation.
* Added a "license" message to the image used.

This is likely it for me for today.

Also added a dozen GitHub issues with the `1.0` tag and a due date. They should keep me busy for the next two days.

@deathbaba, @sompylasar : PTAL.

deathbaba commented:
You need to rebase it first. LGTM.

sompylasar commented:
@dkorolev Is this OK to use the Bricks `Printf` (capitalized) without a `using` or other note in the docu examples?

sompylasar commented on net/api/api.h:
```diff
@@ -91,4 +91,6 @@ inline decltype(Singleton<HTTP_IMPL>()(std::declval<T1>(), std::declval<T2>()))
 }  // namespace net
 }  // namespace bricks
 
+using bricks::net::api::HTTP;
```
How about `GET`, `POST` and other members of `bricks::net::api`? https://github.com/dkorolev/Bricks/blob/3ac9345832934c8d9b61c7bb91b17a1ee6d59950/net/api/types.h#L70

sompylasar replied:
Resolved, you put them into the files where they are defined.

sompylasar commented on net/api/docu/client/docu_02httpclient_01_test.cc:
```diff
@@ -55,8 +54,9 @@ const auto response = HTTP(GET(Printf("localhost:%d/ok", FLAGS_docu_net_client_p
 #endif
-  EXPECT_EQ(200, static_cast<int>(response.code));
   EXPECT_EQ("OK", response.body);
+  EXPECT_EQ(200, static_cast<int>(response.code));    // Show the code as a number if wrong.
```
Show the code as a number **if wrong**. ?

sompylasar commented:
LGTM.

P.S. @dkorolev You've started using GitHub Issues as a task tracker! Thanks :)

dkorolev commented:
Fixed the `Printf` and removed the `if wrong` comment.

(It made quite some sense, but you're right: it would be a bit too long to explain).

@deathbaba : I'm not a fan of rebase, would rather keep the commits as they are.
@sompylasar : PTAL.

sompylasar commented:
@dkorolev :+1: but the README.md didn't get regenerated wrt "if wrong".

dkorolev commented:
Just did.

dkorolev commented:
I'm merging this change now, but just discovered that the `graph` test is broken (`gnuplot` one is doing file, while PNG-s generated by `plotutils` are different).

Guess I'd have to separate unit test (keep it SVG) and docu generation (make it PNG). Will do in a separate pull request.
